image load: no args required

vrothberg opened the pull request:
Disable the args requirement of `image load`.  Instead of requiring a
lower bound, we really need an upper one with at most 1 argument.
Extend the system tests to prevent future regressions.

Fixes: #6718
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

rhatdan commented:
LGTM

edsantiago commented on test/system/120-load.bats:
```diff
@@ -57,9 +57,14 @@ verify_iid_and_name() {
 
-    # Same as above, using stdin
+    # Also make sure that `image load` behaves the same.
+    run_podman image load -i $archive
```
LGTM but wouldn't a better test be to run `image load < $archive` ?

vrothberg replied:
I wanted to use the reproducer from the issue. That's effectively the same code path afaics.

mheon commented:
Is there already a test for at most one argument in the code?

On Tue, Jun 23, 2020, 08:36 Valentin Rothberg <notifications@github.com>
wrote:

> *@vrothberg* commented on this pull request.
> ------------------------------
>
> In test/system/120-load.bats
> <https://github.com/containers/libpod/pull/6722#discussion_r444188306>:
>
> >
> -    # Same as above, using stdin
> +    # Also make sure that `image load` behaves the same.
> +    run_podman image load -i $archive
>
> I wanted to use the reproducer from the issue. That's effectively the same
> code path afaics.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/containers/libpod/pull/6722#discussion_r444188306>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB3AOCH7DKXZ4T67XZG2WPLRYCOTFANCNFSM4OFO57UQ>
> .
>

vrothberg commented:
> Is there already a test for at most one argument in the code?
> […](#)
> On Tue, Jun 23, 2020, 08:36 Valentin Rothberg ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In test/system/120-load.bats <[#6722 (comment)](https://github.com/containers/libpod/pull/6722#discussion_r444188306)>: > - # Same as above, using stdin + # Also make sure that `image load` behaves the same. + run_podman image load -i $archive I wanted to use the reproducer from the issue. That's effectively the same code path afaics. — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <[#6722 (comment)](https://github.com/containers/libpod/pull/6722#discussion_r444188306)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB3AOCH7DKXZ4T67XZG2WPLRYCOTFANCNFSM4OFO57UQ> .

Added one. Also added another test as suggested by @edsantiago to use `image load < $archive`

TomSweeneyRedHat commented:
LGTM
assuming happy tests

edsantiago commented:
>Is there already a test for at most one argument in the code?

That's something I've long wanted to add to `015-help.bats` but have been stymied by inconsistent errors. Given a `podman CMD --help` usage message with no `...` (ellipsis) in the output, it should be possible to count the positional args, and run `podman CMD x y z` with one extra arg. Unfortunately there were, last I checked, too many instances where the error message is unhelpful:
```console
$ ./bin/podman attach x x
Error: `podman attach` requires a name, id  or the "--latest" flag
$ ./bin/podman build x x
Error: context must be a directory: "/path/to/x"
$ ./bin/podman commit x y z
Error: no container with name or ID x found: no such container
```

rhatdan commented:
/lgtm
